### PR TITLE
*: provide a option to wait for init stats to finish before providing service during startup

### LINKED ISSUE #43385: optimizer may make bad decisions due to pseudo stats during init stats
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Here is a small example to demonstrate the issue.

I add `time.Sleep(20 * time.Minute)` into `(*Handle).InitStats` to simulate slow init stats. 
```
mysql> create table t1(a int, b int, index idx(a));
Query OK, 0 rows affected (0.02 sec)

-- insert data into t1

mysql> analyze table t1;
Query OK, 0 rows affected, 1 warning (0.05 sec)
```
Then restart the tidb-server.
```
mysql> explain select * from t1 where a > 1;
+-------------------------+----------+-----------+---------------+--------------------------------+
| id                      | estRows  | task      | access object | operator info                  |
+-------------------------+----------+-----------+---------------+--------------------------------+
| TableReader_7           | 3333.33  | root      |               | data:Selection_6               |
| └─Selection_6           | 3333.33  | cop[tikv] |               | gt(test.t1.a, 1)               |
|   └─TableFullScan_5     | 10000.00 | cop[tikv] | table:t1      | keep order:false, stats:pseudo |
+-------------------------+----------+-----------+---------------+--------------------------------+
3 rows in set (0.00 sec)

mysql> show stats_meta;
Empty set (0.00 sec)

mysql> show stats_histograms;
Empty set (0.00 sec)
```

### 2. What did you expect to see? (Required)
We hope the optimizer avoids using pseudo stats to make decisions. For example, tidb can choose to wait for init stats to finish before providing service.

### 3. What did you see instead (Required)
The optimizer uses pseudo stats during init stats, which may lead to bad plans.

### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by xuyifangreeneyes**
### What problem does this PR solve?

Issue Number: close #43385 ref #42160

Problem Summary:

### What is changed and how it works?

Introduce the configuration `force-init-stats`, which indicates whether to wait for init stats to finish before providing service during startup. 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

Add `time.Sleep(30 * time.Second)` at the beginning of init stats to simulate slow init stats. Start the tidb node with setting `force-init-stats` to true. The connection from mysql client is accepted after init stats is finished(after roughly 30 sec).

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: domain/domain.go [hunk 1]
```diff
@@ -2176,6 +2176,7 @@ func (do *Domain) loadStatsWorker() {
 	} else {
 		logutil.BgLogger().Info("init stats info time", zap.Bool("lite", liteInitStats), zap.Duration("take time", time.Since(t)))
 	}
+	close(statsHandle.InitStatsDone)
```
Comment (qw4990) [inline domain/domain.go:2179]: Better to add this line to the above defer function.

**Review by qw4990 — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**